tests/test_dse.py

```diff
@@ -261,245 +262,371 @@ def test_time_dependent_split(dse, dle):
-
-    Examples
-    --------
-    Instead of the following:
-
-        t0 = a[x, y]
-        t1 = b[x, y]
-        t2 = a[x+1, y+1]*b[x, y]
-        out = t0 + t1 + t2  # pseudocode
-
-    We should get:
-
-        t0 = a[x, y]
-        t1 = b[x, y]
-        out = t0 + t1 + t0[x+1,y+1]*t1[x, y]  # pseudocode
-    """
-    grid = Grid(shape=(3, 3))
-    x, y = grid.dimensions  # noqa
-
-    u = TimeFunction(name='u', grid=grid)
-    u.data[:] = 1.
-    g = Function(name='g', grid=grid)
-    g.data[:] = 2.
-
-    expr = (cos(g)*cos(g) +
-            sin(g)*sin(g) +
-            sin(g)*cos(g) +
-            sin(g[x + 1, y + 1])*cos(g[x + 1, y + 1]))*u
-
-    op0 = Operator(Eq(u.forward, expr), dse='noop')
-    op1 = Operator(Eq(u.forward, expr), dse='aggressive')
-
-    # We expect two temporary Arrays, one for `cos(g)` and one for `sin(g)`
-    arrays = [i for i in FindSymbols().visit(op1) if i.is_Array]
-    assert len(arrays) == 2
-    assert all(i._mem_heap and not i._mem_external for i in arrays)
-    # Check numerical output
-    op0(time_M=1)
-    exp = np.copy(u.data[:])
-    u.data[:] = 1.
-    op1(time_M=1)
-    assert np.allclose(u.data, exp.data, rtol=10e-7)
-
-
-@patch("devito.dse.rewriters.AdvancedRewriter.MIN_COST_ALIAS", 1)
-def test_aliases_different_nests():
-    """
-    Check that aliases arising from two sets of equations A and B,
-    characterized by a flow dependence, are scheduled within A's and B's
-    loop nests respectively.
-    """
-    grid = Grid(shape=(3, 3, 3))
-    x, y, z = grid.dimensions  # noqa
-    t = grid.stepping_dim
-    i = Dimension(name='i')
-
-    f = Function(name='f', grid=grid)
-    f.data_with_halo[:] = 1.
-    g = Function(name='g', shape=(3,), dimensions=(i,))
-    g.data[:] = 2.
-    u = TimeFunction(name='u', grid=grid, space_order=3)
-    v = TimeFunction(name='v', grid=grid, space_order=3)
-
-    # Leads to 3D aliases
-    eqns = [Eq(u.forward, ((u[t, x, y, z] + u[t, x+1, y+1, z+1])*3*f +
-                           (u[t, x+2, y+2, z+2] + u[t, x+3, y+3, z+3])*3*f + 1)),
-            Inc(u[t+1, i, i, i], g + 1),
-            Eq(v.forward, ((v[t, x, y, z] + v[t, x+1, y+1, z+1])*3*u.forward +
-                           (v[t, x+2, y+2, z+2] + v[t, x+3, y+3, z+3])*3*u.forward + 1))]
-    op0 = Operator(eqns, dse='noop', dle=('noop', {'openmp': True}))
-    op1 = Operator(eqns, dse='aggressive', dle=('advanced', {'openmp': True}))
-
-    # Check code generation
-    assert 'bf0' in op1._func_table
-    assert 'bf1' in op1._func_table
-    trees = retrieve_iteration_tree(op1._func_table['bf0'].root)
-    assert len(trees) == 2
-    assert trees[0][-1].nodes[0].body[0].write.is_Array
-    assert trees[1][-1].nodes[0].body[0].write is u
-    trees = retrieve_iteration_tree(op1._func_table['bf1'].root)
-    assert len(trees) == 2
-    assert trees[0][-1].nodes[0].body[0].write.is_Array
-    assert trees[1][-1].nodes[0].body[0].write is v
+class TestAliases(object):
```

Comment:
I'm gonna mark below the new tests. The big diff is only because I've grouped together all aliases tests inside ``TestAliases``